lightning/src/onion_message/messenger.rs

```diff
@@ -614,7 +619,7 @@ where
 			let our_node_id = node_signer.get_node_id(Recipient::Node)
 				.map_err(|()| SendError::GetNodeIdFailed)?;
 			if introduction_node_id == our_node_id {
-				advance_path_by_one(blinded_path, node_signer, &secp_ctx)
+				advance_path_by_one(blinded_path, node_signer, node_id_lookup, &secp_ctx)
 					.map_err(|()| SendError::BlindedPathAdvanceFailed)?;
```

Comment:
`SendError::BlindedPathAdvanceFailed` docs could use an update after this. Looks like we could possibly get rid of that variant and return a `SendError` from `advance_path` but I didn't check too closely.

Reply:
Not sure what you mean by get rid of the variant. What variant would we return here and possibly in `advance_path_by_one`?

Reply:
`advance_path_by_one` has several different errors that it returns, and at a glance it looked like they may be covered by existing `SendError` variants, e.g. `next_hop_pubkey` failure is covered by `SendError::Secp256k1`. But as I said I didn't look too closely so a docs update for `BlindedPathAdvanceFailed` is also fine by me.

Reply:
Ah, yeah, we'd need `NodeSigner::ecdh` to return `SendError::Secp256k1`. But the `KeysManager` implementation returns `()` when given a `Recipient::PhantomNode`.

Reply:
I think any errors returned by the signer need to be a bit more general than `Secp256k1` to cover the case where we have a remote signer that times out, or indeed the phantom case. One option would be to replace `SendError::GetNodeIdFailed` with a more generic `SignerFailed` variant, but that's not related to this PR at all so probably not worth it here.